### ISSUE TITLE
MASTER training

### ISSUE THREAD

**Issue by fracav** [mindee/doctr#331]
Hello  and good morning,

I'm trying your fantastic repo to try different OCR architectures, I have successfully managed to train both the crnn model (in its various versions) and also the SAR model. I am trying to train the MASTER model, with no great results. I get all the time Validation loss: + -3.72135 (Exact: 0.00% | Partial: 0.00%)

I am using a multilingual dataset with about 250k training images and with the other architectures I managed to get to Exact about 85%. (with this specific alphabet  ('3K}7eé;5àÎYho]QwV6qU~W"XnbBvcADfËmy.9ÔpÛ{CôïE%M4#ÈR:g@T$x?0î£|za1ù8,OG€P-''kçHëÀÂ2É/ûIJ\'j(LNÙFut[)èZs+&°Sd=Ï!<â_Ç>r i`l'))  

I am using the doctr/reference/recognition/train.py script. Any suggestions?

**Comment by charlesmindee** [comment]
Hi @fracav, 

Thank you for your report & the attention you pay to doctr.
Concerning the MASTER model, the implementation is very recent and we are training the model for the first time this week. I already spotted a mistake in the encoding process of sequences, and there may be more of them since it is not very stable yet. We will include support for Pytorch within a few days, and finish debugging/training the architecture this week, I invite you to wait 1 week or 2 to use this model if you want work on a definitely stable version!

**Comment by fracav** [comment]
Hi @charlesmindee,

thanks for your very fast response! Hope to try the new architecture soon and good luck with debugging :).
Have a good day. See you soon.

**Comment by charlesmindee** [comment]
This should be fixed now, since #335.
Can you confirm @fracav ?

**Comment by fracav** [comment]
dear @charlesmindee i am glad you solved it quickly. 
Tomorrow I plan to try, I'll let you know soon.

**Comment by charlesmindee** [comment]
The model converged with a batch size of 64 and a learning rate of 1e-4 on my GPU, in the paper they used a learning rate of 1e-4 x number of GPUs, and I believe it is important to respect that: I first tried with a learning rate of 4e-4 and it resulted in very poor results.

**Comment by fracav** [comment]
Hi @charlesmindee  and good morning,

I'm running my experiments on Colab pro (Tesla P100) with input size (48, 160,3), french vocabulary with also whitespace, lr = 0.001 and batch_size = 64. I am using the doctr/references/recognition/train.py script. Over the weekend I was not able to get good results, the predictions are always empty or the repetition of a character max_length times.
Now i will try to fix the lr, do you have any other suggestions?

Thanks in advance

**Comment by charlesmindee** [comment]
Hi @fracav,
I don't have this problem anymore since Friday, I have an input size of 32x128, and I don't use whitspace in the vocab. Otherwise I think we have the same parameters, so if you work on the last commit I guess this could either be a vocabulary of dataset issue ? 
What is the size of your dataset ?

**Comment by fracav** [comment]
I have checked and am using the latest version of the repo.I am using a multilingual dataset (French, German, Italian and some Spanish) of about 250k images. I have now launched a new training with the correct lr and now the training works (after the first epoch of training approximately 53% exact). Thanks for the help. Can you tell me why this model is so sensitive to the chosen lr?

**Comment by charlesmindee** [comment]
Sounds great! I think the huge number of parameters in a transformer model makes it very sensitive to the learning rate, as described [here](https://towardsdatascience.com/hyperparameter-optimization-for-optimum-transformer-models-b95a32b70949): 

"two of the most critical hyperparameters to consider when training a Transformer model on an NLP task is the learning rate and the number of training epochs.

Training for too many epochs or using too high a learning rate typically leads to catastrophic forgetting with the model often resorting to generating the same output/label to any given input. On the other hand, an insufficient number of training epochs or too low a learning rate results in a sub-par model."

**Comment by fracav** [comment]
thanks for the explanation, very interesting. Soon I plan to experiment with handwriting recognition with these architectures. I keep you updated.
Good day.

**Comment by charlesmindee** [comment]
Ok, I will be very interested and curious to see the results!
I am closing this issue for now since it is resolved, but don't hesitate to open a new one if necessary, and feel free to contribute to the package if you want to add a feature or share a dataset!
Have a good day
